code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -238,18 +238,16 @@
 	desc = "Either Nanotrasen's water supply is contaminated, or this machine actually vends lemon, chocolate, and cherry snow cones."
 	list_reagents  = list(/datum/reagent/consumable/ice = 25, /datum/reagent/liquidgibs = 5)
 
-/obj/item/reagent_containers/food/drinks/mug/ // parent type is literally just so empty mug sprites are a thing
+/obj/item/reagent_containers/food/drinks/mug // parent type is literally just so empty mug sprites are a thing
 	name = "mug"
 	desc = "A drink served in a classy mug."
 	icon_state = "tea"
 	inhand_icon_state = "coffee"
 	spillable = TRUE
 
-/obj/item/reagent_containers/food/drinks/mug/on_reagent_change(changetype)
-	if(reagents.total_volume)
-		icon_state = "tea"
-	else
-		icon_state = "tea_empty"
+/obj/item/reagent_containers/food/drinks/mug/on_reagent_change(datum/reagents/reagents)
+	icon_state = reagents.total_volume ? "tea" : "tea_empty"
+	return NONE
```

Comment:
Give all the procs that are called from signals the `SIGNAL_HANDLER` macro, and make their parameters match what is being sent by the signal, please.

Reply:
I can fix the args but as this is an override of `/obj/item/reagent_containers/on_reagent_change` the macro already applies. I also cannot add the macro to the proc body because it can only be applied at the beginning of the base proc.

Reply:
Might have to call parent then.

Reply:
I have shunted all of the icon related code into `update_icon()` and friends. `on_reagent_change()` now just acts to eat args and relay to `update_icon()`. Is this acceptable? I ended up shoveling some non-icon related code in with the rest that was rather inextricable.

Reply:
I've pulled the non-icon related code back out. There's some duplicated code as a result, but whatever. I'm not interested in overhauling how reagent containers update their appearances at the moment.

Reply:
For now this will do. I'll fix the rest of it once #55468 is merged